flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -0,0 +1,343 @@
+		logger.info("Trying to start actor system at {}", hostPortUrl);
+
+		try {
+			Config akkaConfig = AkkaUtils.getAkkaConfig(
+				configuration,
+				new scala.Some<>(new scala.Tuple2<String, Object>(listeningAddress, listeningPort))
+			);
+
+			logger.debug("Using akka configuration\n {}", akkaConfig);
+
+			ActorSystem actorSystem = AkkaUtils.createActorSystem(akkaConfig);
+			logger.info("Actor system started at {}", hostPortUrl);
+			return actorSystem;
+		}
+		catch (Throwable t) {
+			if (t instanceof org.jboss.netty.channel.ChannelException) {
+				Throwable cause = t.getCause();
+				if (cause != null && t.getCause() instanceof java.net.BindException) {
+					throw new IOException("Unable to create ActorSystem at address " + hostPortUrl +
+							" : " + cause.getMessage(), t);
+				}
+			}
+			throw new Exception("Could not create actor system", t);
+		}
+	}
+
+	/**
+	 * Starts the web frontend.
+	 * @param config The Flink config.
+	 * @param actorSystem The ActorSystem to start the web frontend in.
+	 * @param logger Logger for log output
+	 * @return WebMonitor instance.
+	 * @throws Exception
+	 */
+	public static WebMonitor startWebMonitorIfConfigured(
+				Configuration config,
+				ActorSystem actorSystem,
+				ActorRef jobManager,
+				Logger logger) throws Exception {
+
+
+		// this ensures correct values are present in the web frontend
+		final Address address = AkkaUtils.getAddress(actorSystem);
+		config.setString(ConfigConstants.JOB_MANAGER_IPC_ADDRESS_KEY, address.host().get());
+		config.setString(ConfigConstants.JOB_MANAGER_IPC_PORT_KEY, address.port().get().toString());
+
+		if (config.getInteger(ConfigConstants.JOB_MANAGER_WEB_PORT_KEY, 0) >= 0) {
+			logger.info("Starting JobManager Web Frontend");
+
+			LeaderRetrievalService leaderRetrievalService = 
+				LeaderRetrievalUtils.createLeaderRetrievalService(config, jobManager);
+
+			// start the web frontend. we need to load this dynamically
+			// because it is not in the same project/dependencies
+			WebMonitor monitor = WebMonitorUtils.startWebRuntimeMonitor(
+				config, leaderRetrievalService, actorSystem);
+
+			// start the web monitor
+			if (monitor != null) {
+				String jobManagerAkkaURL = AkkaUtils.getAkkaURL(actorSystem, jobManager);
+				monitor.start(jobManagerAkkaURL);
+			}
+			return monitor;
+		}
+		else {
+			return null;
+		}
+	}
+
+	/**
+	 * Generate a task manager configuration.
+	 * @param baseConfig Config to start from.
+	 * @param jobManagerHostname Job manager host name.
+	 * @param jobManagerPort Port of the job manager.
+	 * @param numSlots Number of slots to configure.
+	 * @param registrationTimeout Timeout for registration
+	 * @return TaskManager configuration
+	 */
+	public static Configuration generateTaskManagerConfiguration(
+				Configuration baseConfig,
+				String jobManagerHostname,
+				int jobManagerPort,
+				int numSlots,
+				FiniteDuration registrationTimeout) {
```

Comment:
I was just wondering because I noted that you did it differently in different source files.